packages/flutter/lib/src/material/app.dart

```diff
@@ -201,8 +202,58 @@ class MaterialApp extends StatefulWidget {
        assert(checkerboardOffscreenLayers != null),
        assert(showSemanticsDebugger != null),
        assert(debugShowCheckedModeBanner != null),
+       routeNameParser = null,
+       routerDelegate = null,
+       backButtonDispatcher = null,
+       routeNameProvider = null,
        super(key: key);
 
+  /// Creates a MaterialApp with router.
+  const MaterialApp.router({
+    Key key,
+    @required this.routeNameParser,
+    @required this.routerDelegate,
+    this.backButtonDispatcher,
+    this.routeNameProvider,
+    this.builder,
+    this.title = '',
+    this.onGenerateTitle,
+    this.color,
+    this.theme,
+    this.darkTheme,
+    this.themeMode = ThemeMode.system,
+    this.locale,
+    this.localizationsDelegates,
+    this.localeListResolutionCallback,
+    this.localeResolutionCallback,
+    this.supportedLocales = const <Locale>[Locale('en', 'US')],
+    this.debugShowMaterialGrid = false,
+    this.showPerformanceOverlay = false,
+    this.checkerboardRasterCacheImages = false,
+    this.checkerboardOffscreenLayers = false,
+    this.showSemanticsDebugger = false,
+    this.debugShowCheckedModeBanner = true,
+    this.shortcuts,
+    this.actions,
+  }) : assert(routeNameParser != null),
+      assert(routerDelegate != null),
+      assert(title != null),
+      assert(debugShowMaterialGrid != null),
+      assert(showPerformanceOverlay != null),
+      assert(checkerboardRasterCacheImages != null),
+      assert(checkerboardOffscreenLayers != null),
+      assert(showSemanticsDebugger != null),
+      assert(debugShowCheckedModeBanner != null),
+      navigatorObservers = null,
+      navigatorKey = null,
+      onGenerateRoute = null,
+      home = null,
+      onGenerateInitialRoutes = null,
+      onUnknownRoute = null,
```

Comment:
We follow the [final_not_initialized_constructor](https://dart.dev/tools/diagnostic-messages#final_not_initialized_constructor)

Reply:
Ohh, I am confused because not found  [final_not_initialized_constructor](https://dart.dev/tools/diagnostic-messages#final_not_initialized_constructor) in https://github.com/flutter/flutter/blob/master/analysis_options.yaml